airflow/contrib/example_dags/example_gcp_sql_query.py

```diff
@@ -0,0 +1,322 @@
+    "mysql://{user}:{password}@127.0.0.1:{proxy_port}/{database}".format(
+        user=MYSQL_USER,
+        password=MYSQL_PASSWORD,
+        proxy_port=MYSQL_PROXY_PORT,
+        database=MYSQL_DATABASE_NAME)
+
+MYSQL_PROXY_SOCKET_PATH = "/tmp/gcsp_sockets.mysql_socket_short_id/" \
+                          "{project_id}:{location}:{instance}".format(
+                              project_id=PROJECT_ID,
+                              location=LOCATION,
+                              instance=MYSQL_INSTANCE_NAME
+                          )
+
+os.environ['AIRFLOW_CONN_PROXY_MYSQL_SOCKET'] = \
+    "mysql://{user}:{password}@localhost/{database}?unix_socket={quoted_socket_path}"\
+    .format(
+        user=MYSQL_USER,
+        password=MYSQL_PASSWORD,
+        database=MYSQL_DATABASE_NAME,
+        quoted_socket_path=quote_plus(MYSQL_PROXY_SOCKET_PATH)
+    )
+
+os.environ['AIRFLOW_CONN_PUBLIC_MYSQL_TCP'] = \
+    "mysql://{user}:{password}@{public_ip}:{public_port}/{database}"\
+    .format(
+        user=MYSQL_USER,
+        password=MYSQL_PASSWORD,
+        public_ip=MYSQL_PUBLIC_IP,
+        public_port=MYSQL_PUBLIC_PORT,
+        database=MYSQL_DATABASE_NAME)
+
+SSL_SPEC = {
+    'cert': MYSQL_CLIENT_CERT_FILE,
+    'key': MYSQL_CLIENT_KEY_FILE,
+    'ca': MYSQL_SERVER_CA_FILE
+}
+
+os.environ['AIRFLOW_CONN_PUBLIC_MYSQL_TCP_SSL'] = \
+    "mysql://{user}:{password}@{public_ip}:{public_port}/{database}?" \
+    "ssl={ssl_spec}"\
+    .format(
+        user=MYSQL_USER,
+        password=MYSQL_PASSWORD,
+        public_ip=MYSQL_PUBLIC_IP,
+        public_port=MYSQL_PUBLIC_PORT,
+        database=MYSQL_DATABASE_NAME,
+        ssl_spec=quote_plus(json.dumps(SSL_SPEC))
+    )
+
+# [END howto_operator_cloudsql_query_connections]
+
+# [START howto_operator_cloudsql_query_operators]
+with models.DAG(
+    dag_id='example_gcp_sql_query',
+    default_args=default_args,
+    schedule_interval=None
+) as dag:
+    sql_instance_deploy_task_postgres_proxy_tcp = CloudSqlQueryOperator(
+        task_id="example_gcp_sql_task_postgres_proxy_tcp",
+        sql=SQL,
+        project_id=PROJECT_ID,
+        location=LOCATION,
+        instance=POSTGRES_INSTANCE_NAME,
+        database_type='postgres',
+        use_proxy=True,
+        sql_proxy_use_tcp=True,
+        sql_proxy_tcp_port=POSTGRES_PROXY_PORT,
+        sql_proxy_force_download=False,
+        cloudsql_conn_id='proxy_postgres_tcp',
+        database=POSTGRES_DATABASE_NAME
+    )
+    sql_instance_deploy_task_postgres_proxy_socket = CloudSqlQueryOperator(
+        task_id="postgres_socket_short_id",
+        sql=SQL,
+        project_id=PROJECT_ID,
+        location=LOCATION,
+        instance=POSTGRES_INSTANCE_NAME,
+        database_type='postgres',
+        use_proxy=True,
+        sql_proxy_use_tcp=False,
+        sql_proxy_force_download=False,
+        cloudsql_conn_id='proxy_postgres_socket',
+        database=POSTGRES_DATABASE_NAME,
+        dag=dag
```

Comment:
I think if we use `with models.DAG` it's not necessary to pass `dag=dag` explicitly to the operator.